appserver/payara-appserver-modules/jmx-monitoring-service/src/main/java/fish/payara/jmx/monitoring/admin/SetJMXMonitoringConfiguration.java

```diff
@@ -306,8 +306,7 @@ private void updateAttributes(MonitoringServiceConfiguration monitoringConfig, A
      * attribute value.
      */
     private MonitoredAttribute parseToMonitoredAttribute(String input, MonitoredAttribute monitoredAttribute) throws PropertyVetoException {
-        // Negative lookbehind - find space characters not preceeded by \
-        String[] attributeTokens = input.split("(?<!\\\\) ");
+        String[] attributeTokens = input.split("(?=objectName)|(?=attributeName)|(?=description)");
```

Comment:
This is a neat fix, but unfortunately has a small hole in it.

If you use the command:
~~~
set-monitoring-configuration --addattribute "objectName=myObject attributeName=myAttribute description=a test attribute"
~~~

the stored `attributeName` will be `myAttribute  ` (with a space).

If you then run:

~~~
set-monitoring-configuration --delattribute "objectName=myObject attributeName=myAttribute"
~~~

the attribute to be searched for will be `myAttribute` (without a space).

The parsed string will need to be `String.trim()`ed probably [here](https://github.com/MeroRai/Payara/blob/b69e1071adb7b4959ead91fe05ff53f101775514/appserver/payara-appserver-modules/jmx-monitoring-service/src/main/java/fish/payara/jmx/monitoring/admin/SetJMXMonitoringConfiguration.java#L331).

Reply:
This regex string also has the problem of hating when the description has the word 'description' in. I would recommend something like changing the regex string to '(?=objectName ?=)|(?=attributeName ?=)|(?=description ?=)' to make sure it catches this!